src/threading.c

```diff
@@ -49,6 +49,21 @@ JL_DLLEXPORT _Atomic(uint8_t) jl_measure_compile_time_enabled = 0;
 JL_DLLEXPORT _Atomic(uint64_t) jl_cumulative_compile_time = 0;
 JL_DLLEXPORT _Atomic(uint64_t) jl_cumulative_recompile_time = 0;
 
+JL_DLLEXPORT uint64_t jl_thread_up_time(void)
+{
+    jl_ptls_t ptls = jl_current_task->ptls;
+    return jl_record_time_for_tls_metric() - ptls->timing_tls.start_time;
+}
+
+JL_DLLEXPORT uint64_t jl_thread_user_time(uint8_t tid)
+{
+    jl_ptls_t ptls = jl_atomic_load_relaxed(&jl_all_tls_states)[tid];
+    jl_timing_tls_states_t *timing = &ptls->timing_tls;
+    return jl_record_time_for_tls_metric() - timing->start_time -
```

Comment:
+1 to your suggestion, Nick.

I think the right thing to do will be to do this math in DataDog, and subtract these out of the kernel CPU time metric we get from Kubernetes.

Open to other directions, but that's where my current thinking is at as well.

Reply:
> think about how these numbers come out when we're oversubscribed

Why oversubscription would cause any issues with these metrics?

Reply:
> Why oversubscription would cause any issues with these metrics?

they might not, but i worry they might because all threads will have the global start_time, but then some threads won't be on the cpu and idk if our scheduler_time (assuming this includes sleep time) is going to capture that?

Reply:
https://github.com/RelationalAI/julia/pull/181#discussion_r1781499757